sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,653 @@
+
+    Returns
+    -------
+    (doc_topic_distr, component_diff) :
+        `doc_topic_distr` is unnormailzed topic distribution for each document.
+        In literature, it is `gamma` latent variable. we can calcuate `E[log(theta)]`
+        from it.
+        `component_diff` is the difference of `component_` when `cal_diff` is True.
+        Otherwise, it is None.
+
+    """
+
+    n_samples, n_features = X.shape
+    n_topics = exp_topic_word_distr.shape[0]
+
+    # this is variable `gamma` in literature
+    doc_topic_distr = np.ones((n_samples, n_topics))
+    # this is `exp(E[log(theta)])` in literature
+    exp_doc_topic = np.exp(_dirichlet_expectation(doc_topic_distr))
+
+    # diff on `component_` (only calculate it when `cal_diff` is True)
+    component_diff = np.zeros(exp_topic_word_distr.shape) if cal_diff else None
+
+    X_data = X.data
+    X_indices = X.indices
+    X_indptr = X.indptr
+
+    for d in xrange(n_samples):
+        ids = X_indices[X_indptr[d]:X_indptr[d + 1]]
+        cnts = X_data[X_indptr[d]:X_indptr[d + 1]]
+
+        doc_topic_d = doc_topic_distr[d, :]
+        exp_doc_topic_d = exp_doc_topic[d, :]
+        exp_topic_word_d = exp_topic_word_distr[:, ids]
+
+        # The optimal phi_{dwk} is proportional to 
+        # exp(E[log(theta_{dk})]) * exp(E[log(beta_{dw})]).
+        norm_phi = np.dot(exp_doc_topic_d, exp_topic_word_d) + 1e-100
+
+        # Iterate between `doc_topic_d` and `norm_phi` until convergence
+        for it in xrange(0, max_iters):
+            last_d = doc_topic_d
+
+            doc_topic_d = doc_topic_prior + exp_doc_topic_d * \
+                np.dot(cnts / norm_phi, exp_topic_word_d.T)
+            exp_doc_topic_d = np.exp(_dirichlet_expectation(doc_topic_d))
+            norm_phi = np.dot(exp_doc_topic_d, exp_topic_word_d) + 1e-100
+
+            meanchange = mean_change(last_d, doc_topic_d)
+            if meanchange < mean_change_tol:
+                break
+        doc_topic_distr[d, :] = doc_topic_d
+
+        # Contribution of document d to the expected sufficient
+        # statistics for the M step.
+        if cal_diff:
+            component_diff[:, ids] += np.outer(exp_doc_topic_d, cnts / norm_phi)
+
+    return (doc_topic_distr, component_diff)
+
+
+class LatentDirichletAllocation(BaseEstimator, TransformerMixin):
+
+    """
+    Online Latent Dirichlet Allocation implementation with variational inference
+
+    References
+    ----------
+    [1] "Online Learning for Latent Dirichlet Allocation", Matthew D. Hoffman, 
+        David M. Blei, Francis Bach
+
+    [2] Matthew D. Hoffman's onlineldavb code. Link:
+        http://www.cs.princeton.edu/~mdhoffma/code/onlineldavb.tar
+
+
+    Parameters
+    ----------
+    n_topics : int, optional (default: 10)
+        Number of topics.
+
+    doc_topic_prior : float, optional (defalut: 0.1)
+        Prior of document topic distribution `theta`. In general, it is `1 / n_topics`.
```

Comment:
I'm not sure if "in general" is a good formulation. Shouldn't this default be coupled to the default of ``n_topics``, so that is someone changes ``n_topics`` the ``doc_topic_prior`` also changes?